tests/unit/llms/test_llm_loader.py

```diff
@@ -105,3 +105,22 @@ def test_constructor_when_missing_model_config():
     message = f"No configuration provided for model {test_model} under LLM provider {test_provider}"
     with pytest.raises(ModelConfigMissingError, match=message):
         LLMLoader(provider=test_provider, model=test_model)
+
+
+providers = [
+    constants.PROVIDER_OPENAI,
+    constants.PROVIDER_BAM,
+]
+
+
+@pytest.mark.parametrize("provider", providers)
+def test_constructor_no_keys(provider):
```

Comment:
Not sure what it is testing. That there is a raise on config when no API keys are configured?

Reply:
yes. there's some `except->re raise` blocks we need to "touch"